### PR TITLE
[App Service] `az staticwebapp create`: Add better error message when attempting to create a static webapp that already exists

### PR DESCRIPTION
**Pull request by StrawnSC**
**Description**

Previously, trying to create a static webapp that already exists results in a `Bad Request` error that doesn't contain useful information even with `--debug`. It also fails at the very end of the command execution (after the user already completed the interactive github validation process to get a token). This PR changes `az staticwebapp create` to first validate that the app with the given name/RG does not already exist  

**Testing Guide**

`azdev test test_create_validator`

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

**Review by wangzelin007 — APPROVED** [review]
LGTM

File: src/azure-cli/azure/cli/command_modules/appservice/_validators.py [hunk 1]
```diff
@@ -246,6 +246,22 @@ def _validate_service_tag_existence(cmd, namespace):
...
+
+    client = get_mgmt_service_client(cmd.cli_ctx, WebSiteManagementClient).static_sites
+    try:
+        client.get_static_site(namespace.resource_group_name, namespace.name)
+        exists = True
+    except ResourceNotFoundError:
+        exists = False
+
+    if exists:
+        raise ValidationError("Static Web App {} already exists in resource group {}".format(
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_validators.py:261]: cli create command needs to be idempotent, infact the Create_or_Update APIs should be Idempotent, while I am not sure if this is a regression in the API, while we wait, lets keep this (hopefully) as a temporary fix & ensure that if the app exists we do a GET on the SWA & return that response. Thanks!

**Review by panchagnula — CHANGES_REQUESTED** [review]
Lets make sure to follow CLI's standards to keep the cli create command idempotent.

**Comment by StrawnSC** [comment]
> Lets make sure to follow CLI's standards to keep the cli create command idempotent.

Changed the create command so that it returns the pre-existing site (and warns the user) if they try and run create on a SWA that already exists

**Review by panchagnula — APPROVED** [review]
LGTM - just fix the merge conflicts & get build to green so it can be merged.

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**